docs/docs/concepts/flows.mdx

```diff
@@ -483,6 +483,44 @@ flow is completed.
   scope: "flow"
 ```
 
+#### Slot value validation
+
+You can now write slot value validation directly in your flow yaml file without the need to use custom actions.
+Add a `rejections` property to your `collect_information` step to define the validation rules.
+The `rejections` section is a list of mappings, where each mapping has an `if` and an `utter` property:
+- the `if` property is a condition written in natural language that is evaluated using the
+[pypred](https://github.com/armon/pypred) library.
+- the `utter` property is the name of the response the assistant will send if the condition evaluates to `True`.
+
+Here is an example:
+
+```yaml
+flows:
+  verify_eligibility:
+    description: This flow verifies if the user is eligible for creating an account.
+    name: verify eligibility
+    steps:
+      - id: "ask_age"
+        collect: "age"
+        rejections:
+          - if: age < 1
+            utter: utter_invalid_age
+          - if: age < 13
+            utter: utter_age_at_least_13
+        next: "ask_email"
+```
+
+Upon every validation failure, the assistant will again make another attempt to collect the information.
+This will be repeated until none of the rejection conditions pass. Once all rejections have evaluated to `False`, the
+assistant moves on to the `next` step with the collected value.
+
+If the predicate defined in the `if` property is invalid or missing, the assistant will log an error and respond with the
```

Comment:
I'm a bit confused here. For me there are 3 error cases:
- `if` property missing -> should be detected in `rasa data validate` + runtime error as failsafe using `utter_internal_error_rasa`
- `if` property defines an invalid predicate -> should be detected in `rasa data validate` + runtime error as failsafe using `utter_internal_error_rasa`
- the execution of the predicates raises an error -> runtime error that uses `utter_internal_error_rasa`


Does it work this way?

Reply:
> `if` property missing

I meant if it's explicitly coded as `null` / None in the yaml file. The schema for the yaml format should catch an `if` required property that is missing.
The error cases you listed are correct.